Mathlib/Init/Algebra/Classes.lean

```diff
@@ -177,6 +177,12 @@ class IsAntisymm (α : Type u) (r : α → α → Prop) : Prop where
 class IsTrans (α : Type u) (r : α → α → Prop) : Prop where
   trans : ∀ a b c, r a b → r b c → r a c
 
+instance {α : Type u} {r : α → α → Prop} [IsTrans α r] : Trans r r r :=
+  ⟨IsTrans.trans _ _ _⟩
+
+instance {α : Type u} {r : α → α → Prop} [Trans r r r] : IsTrans α r :=
+  ⟨fun _ _ _ => Trans.trans⟩
+
```

Comment:
Expecting the answer "yes": so these looping instances are OK?

Reply:
@digama0 said yes.